packages/components/src/Avatar/src/Avatar.tsx

```diff
@@ -0,0 +1,243 @@
+     */
+    size: ResponsiveProp<AvatarSize>;
+    /**
+     * The src of the image to display. If not provided, the initials will be displayed instead.
+     */
+    src: string;
+    /**
+    * The src of the image to display if the image fails to load. If set to null, the initials will be displayed instead.
+    * * @default "BrokenImageRichIcon"
+    */
+    fallbackSrc?: string | null;
+}
+
+export interface AvatarProps extends StyledSystemProps, BaseComponentProps, Omit<AvatarImageProps, "size" | "src">, Omit<AvatarInitialsProps, "size"> {
+    /**
+     * The size of the avatar.
+     * * @default "md"
+     */
+    size?: ResponsiveProp<AvatarSize>;
+    /**
+     * The src of the image to display. If not provided, the initials will be displayed instead.
+     */
+    src?: string;
+}
+
+export const AvatarToTextSizeAdapter: SizeAdapter<AvatarProps["size"], TextProps["size"]> = {
+    xs: "xs",
+    sm: "xs",
+    md: "md",
+    lg: "lg",
+    xl: "xl",
+    "2xl": "xl"
+};
+
+function getColorIndex(name: string, maxNumberOfColor: number) {
+    let hashCode = 0;
+
+    for (let i = name.length - 1; i >= 0; i--) {
+        const character = name.charCodeAt(i);
+        const shift = i % 8;
+
+        hashCode ^= (character << shift) + (character >> (8 - shift));
+    }
+
+    return hashCode % maxNumberOfColor;
+}
+
+function getColorName(name: string) {
+    const variantToUse = `option${getColorIndex(name, 8) + 1}`;
+
+    return `decorative-${variantToUse}`;
+}
+
+function AvatarInitials(props: AvatarInitialsProps) {
+    const { 
+        name,
+        size: sizeValue
+    } = props;
+    
+    const size = useResponsiveValue(sizeValue) ?? "md";
+
+    const initials = useMemo(() => {
+        const cleanName = name.replace(/\s+/g, " ").trim();
+        const [firstName, lastName] = cleanName.split(" ");
+        const letters = firstName && lastName
+            ? `${firstName.charAt(0)}${lastName.charAt(0)}`
+            : firstName.charAt(0);
+
+        return size === "xs" && letters.length > 1 ? letters.charAt(0) : letters;
+    }, [name, size]);
+
+    return (
+        <Text
+            size={AvatarToTextSizeAdapter[size]}
+            className={styles["hop-Avatar__initials"]}
+        >
+            {initials}
+        </Text>
+    );
+}
+
+function Avatar(props: AvatarProps, ref: ForwardedRef<HTMLDivElement>) {
+    [props, ref] = useContextProps(props, ref, AvatarContext);
+    const { stylingProps, ...ownProps } = useStyledSystem(props);
+    const {
+        "aria-label": ariaLabel,
+        className,
+        fallbackSrc,
+        imageProps,
+        isDisabled,
+        name,
+        size: sizeValue,
+        slot,
+        src,
+        style,
+        ...otherProps
+    } = ownProps;
+    const domProps = filterDOMProps(otherProps);
+    
+    const { onError, onLoad, ...otherImageProps } = imageProps ?? {};
+    const [imageUrl, status] = useImageFallback({ src, fallbackSrc, onError, onLoad });
```

Comment:
This is how i'd imagine the code everywhere using this hook. Instead of a tuple as a return value, we should return an object
```    
    const [imageUrl, status] = useImageFallback({ src, fallbackSrc, onError, onLoad });
    const imageLoaded = status === "loaded";
    const imageFailed = status === "failed";
```

```
    const { imageUrl, isLoaded, isFailed } = useImageFallback({ src, fallbackSrc, onError, onLoad });
```
you can still return the status in the return object if you want.